pkg/config/config_template.yaml

```diff
@@ -194,8 +194,8 @@ api_key:
 ## trigger security alert when querying unknown IP (for example, when enabling
 ## Tencent on AWS).
 ## Setting an empty list will disable querying any cloud metadata endpoints
-## (falling back on system metadata). Using this setting on a cloud provider
-## host may result in duplicated hosts in your account.
+## (falling back on system metadata). Disabling metadata for the cloud provider in which an Agent runs may result in
+## duplicated hosts in your Datadog account and missing Autodiscovery features
```

Comment:
I don't think this change would affect autodiscovery, by design ECS listeners aren't enabled if the agent isn't running on ECS, I think you just had to make changes in the autodiscovery package because you changed the signature of the`V2()` function.

Reply:
I see. Couldn't disabling the option per se (regardless of my changes) also lead to weird behavior with Autodiscovery? I think that is what @olivielpeau  mentioned above and why I added it.

Reply:
yes I would prefer keeping this in the documentation. btw I'm curious which package has kept calling the ECS api on azure? I don't think it was autodiscovery, right?

Reply:
My guess is the Metadata Collector as it runs on a schedule